std-bits/table/src/main/java/org/enso/table/data/index/MultiValueKey.java

```diff
@@ -71,4 +74,120 @@ protected static Object FoldObject(Object value) {
 
     return value;
   }
+
+  public static int compareObjects(Object thisValue, Object thatValue) throws ClassCastException {
```

Comment:
I'm a bit worried about this approach to comparisons - it will not work at all with custom Enso types which can define a `compare_to`, and moreover we are duplicating the logic of primitive comparisons from interpreter - I feel like this is going to bite us one way or another when we will have inconsistencies.

I think we can have this as a temporary solution if it is going to be revisited when implementing the table ordering components - but I think at some point we should revisit this to allow custom types and somehow avoid duplicating the comparison logic.

It may be worth looking at current `Table.sort` implementation and the `OrderBuilder.java`.

It may also be worth to chat with @kustosz about how we can avoid the overhead of crossing the polyglot boundary too much. My intuition is - for custom types we simply cannot avoid this, but for builtin types we may be able to have specializations of the comparator in the Storages to even avoid boxing them which would give even better performance. Still would be good to somehow avoid duplicating the logic of primitive comparisons between the interpreter and the library - @kustosz do you have any ideas how we could achieve that? Seems a bit like we can _either_ avoid crossing the boundary or re-use the builtin comparisons but not both. But maybe we can figure something out  -  like splitting a separate, self-contained, library which could encapsulate the comparison logic and which could be used both in the interpreter and in the library? I'm not sure if this is not getting too complicated but I also think that duplicating this logic is asking for trouble in the future.